compiler/src/iree/compiler/Codegen/Common/GPU/GPUReduceBankConflicts.cpp

```diff
@@ -18,6 +18,23 @@ namespace mlir::iree_compiler {
 
 namespace {
 
+/// Check if AllocOp has a CollapseShapeOp user.
+static bool hasCollapseShapeUser(memref::AllocOp allocOp) {
+  SmallVector<Operation *> users(allocOp->getUsers());
+  while (!users.empty()) {
+    auto user = users.pop_back_val();
+    if (isa<memref::CollapseShapeOp>(user)) {
+      return true;
+    }
+    if (isa<ViewLikeOpInterface>(user)) {
+      for (auto u : user->getUsers()) {
```

Comment:
Should probably add arith::SelectOp to this too. And ops with a block that has an argument tied to this use (I think we mainly need to worry about scf.for).

Reply:
I think both of these cases should be okay with the pass, as it will add a strided subview with the same shape as the pre padded alloc and replace the type in all the users, the problem with the collapse shape case is it does not support strides that dont add up to the shape.

If there is some IR case you are worried about, happy to add that as a test.